### PR TITLE
first stab at differentiating query key on `useQuery` and `useInfiniteQuery`

### LINKED ISSUE #3128: feat: prefix infinite queries differently to "normal" queries?
### Describe the feature you'd like to request

Currently, the cache gets a bit messed up if you use an endpoint the with `.useInfiniteQuery()` in one part of the app and `.useQuery()` in another.

### Describe the solution you'd like to see

Currently, we store cache keys as `[pathArray, input]`-tuple

Maybe, we should store it as `['query' | 'infinite', pathArray, input]`-tuple?

Or a `[pathArray, 'query' | 'infinite', input]`-tuple?

### Desribe alternate solutions

Not doing it

### Additional information

_No response_

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by KATT**
Closes #3128

## 🎯 Changes

- First stab at #3128
- Changing the react-query query key to `['infiniteQuery' | 'query', path, unknown?]` in order to fix cases where you `getQuery` on a thing that has previously been fetched as `useInfiniteQuery` and vice versa
- Not fully functioning, I don't have time to work on it right now, **help wanted 🙏 🙏 🙏 **

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/next/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Comment by KATT** [comment]
cc @freeatnet @JonParton any of you that would consider lending a hand on this?

**Comment by JonParton** [comment]
I managed to get a start on this on the plane( 708f0f6 ), however it is going pretty deep...

I have also had to alter the approach slightly due to thinking about how things like invalidate may want to act on all queries in the cache and not be filtered to just `queries` or `infiniteQueries`...  I also needed to make sure that what we did worked well with the React query fuzzy matching...

Any ways ... ran out of time and not happy with everything done so far but I'm going to try and come back to this when I have got over the back to work in the morning tidal wave 🌊  ... I might be able to come up with a smarter way!...

**Comment by KATT** [comment]
Hey @JonParton, did you play more with this?

I think the best structure of the tuple will be


```ts
[['post', 'list'], 'infiniteQuery', { limit: 5 } ]
[['post', 'list'], 'query', { limit: 5 } ]
```

Then:

- `post.list.invalidate()` works easily and as expected
- `post.list.setQueryData()` only sets data for `'query'` and same for infinite
- `post.list.getQueryData()` only gets data for `'query'` and same for infinite
- ..

**Comment by JonParton** [comment]
Hey @KATT  - I haven't had time to look at this any more yet but I may do tomorrow subject to the query key structure comment in [the issue comments!](https://github.com/trpc/trpc/issues/3128#issuecomment-1312480321) 👍

**Comment by KATT** [comment]
Oh, I missed that! That structure would work too 👌 

tl;dr, your suggestion something like this

```ts
[ ["post", "byId"], { type: "query", input: { userId:10 } } ]
```

Which is very similar to:

```ts
[ ["post", "byId"], "query", { userId:10 } ]
```

I don't care that much, the extra bytes don't bother me & users will rarely look and care about these structures, do whatever feels best :)

**Comment by JonParton** [comment]
@KATT - I've managed to implement this with a much smaller footprint in #3165 🎉🎉 - Give it a review and if OK we can close this one!
